src/huggingface_hub/hf_api.py

```diff
@@ -992,34 +1014,47 @@ def dataset_info(
 
     def create_repo(
         self,
-        name: str,
+        repo_id: str = None,
         token: Optional[str] = None,
         organization: Optional[str] = None,
         private: Optional[bool] = None,
         repo_type: Optional[str] = None,
         exist_ok=False,
-        lfsmultipartthresh: Optional[int] = None,
         space_sdk: Optional[str] = None,
+        name: str = None,
     ) -> str:
-        """
-        HuggingFace git-based system, used for models, datasets, and spaces.
+        """Create an empty repo on the HuggingFace Hub.
+
+        Args:
+            repo_id: A namespace (user or an organization) and a repo name
+                seperated by a ``/``.
 
-        Call HF API to create a whole repo.
+                .. versionadded: 0.4.0
 
-        Params:
-            private: Whether the model repo should be private (requires a paid huggingface.co account)
+            token: An authentication token [1]_.
 
-            repo_type: Set to :obj:`"dataset"` or :obj:`"space"` if uploading to a dataset or space, :obj:`None` or :obj:`"model"` if uploading to a model. Default is :obj:`None`.
+            private: Whether the model repo should be private.
 
-            exist_ok: Do not raise an error if repo already exists
+            repo_type: Set to :obj:`"dataset"` or :obj:`"space"` if uploading
+                to a dataset or space, :obj:`None` or :obj:`"model"` if
+                uploading to a model. Default is :obj:`None`.
 
-            lfsmultipartthresh: Optional: internal param for testing purposes.
+            exist_ok: If ``True``, do not raise an error if repo already
+                exists.
 
-            space_sdk: Choice of SDK to use if repo_type is "space". Can be "streamlit", "gradio", or "static".
+            space_sdk: Choice of SDK to use if repo_type is "space". Can be
+                "streamlit", "gradio", or "static".
 
         Returns:
             URL to the newly created repo.
+
+        References:
+            .. [1] https://huggingface.co/settings/tokens
+
+        Reference
         """
+        name, organization = _validate_repo_id_deprecation(repo_id, name, organization)
```

Comment:
I would add a check to ensure that at least `repo_id` or `name` is passed, since they are now both optional arguments:
```
>>> from huggingface_hub import create_repo
>>> create_repo()
Traceback (most recent call last):
  File "/usr/lib/python3.6/code.py", line 91, in runcode
    exec(code, self.locals)
  File "<input>", line 1, in <module>
  File "/home/lysandre/Workspaces/Python/huggingface_hub/src/huggingface_hub/hf_api.py", line 1056, in create_repo
    name, organization = _validate_repo_id_deprecation(repo_id, name, organization)
  File "/home/lysandre/Workspaces/Python/huggingface_hub/src/huggingface_hub/hf_api.py", line 68, in _validate_repo_id_deprecation
    if "/" in repo_id:
TypeError: argument of type 'NoneType' is not iterable
```

Something like the following
```suggestion
        if name is None and repo_id is None:
            raise ValueError("Please pass either `name` or `repo_id`")
            
        name, organization = _validate_repo_id_deprecation(repo_id, name, organization)
```

Reply:
+1, but since name is deprecated, I would just ask to specify `repo_id`.

Reply:
Indeed, good call!